### DISCUSSION TITLE
Offset is bigger than the size of data to read in ReadFile

### DISCUSSION THREAD

**Discussion by Thomas-Giaroli** [dokan-dev/dokan-dotnet#336] [Q&A] [answered]
Hello, i have a new problem with my dokan-dotnet based software. I have no idea how this is possible but when opening files, rarely, my offset will end up being bigger than the data i have to read, leaving me with a negative value affected to bytesRead, and of course, throwing an error on my call to Array.copy.

Here is a gist with my readFile function, i have no idea if any other parts of my code would help so don't hesitate to ask me for more :

https://gist.github.com/Thomas-Giaroli/6c1553cce846ff2c5c43b2a4a7b1d026

Thanks already :)

**Comment by Liryna** [comment]
The offset is the start where to read and the length is how much data from this offset to read. You cannot end up with a negative value.

Please look at the native memfs implementation.

**Reply by Thomas-Giaroli** [reply]
I know about the offset and the purpose it serves, it never gets a negative value, but it still gets bigger than my data, which ends up making my bytesRead negative.  I have no idea why this happens.

![image](https://github.com/dokan-dev/dokan-dotnet/assets/72007344/6f52cfe4-3b62-43c2-92e2-0d72b745b66e)

As you can see on this screenshot offset is bigger than "arr", which is my data to read in a byte array.

Resulting in this error :

![image](https://github.com/dokan-dev/dokan-dotnet/assets/72007344/e137c374-72a3-43e7-b471-95a11d80cd72)

**Reply by Liryna** [reply, marked as answer]
Oh I see. In this case, there is a specific error code. You should be able to find it in the memfs sample code.
